recipe_scrapers/_utils.py

```diff
@@ -175,3 +176,25 @@ def change_keys(obj, convert):
     else:
         return obj
     return new
+
+
+class ScraperJSONEncoder(JSONEncoder):
+    """
+    Custom JSONEncoder that can be used to json dump a scraper by:
+    json.dumps(scraper, cls=ScraperJSONEncoder, indent=True)
+    """
+
+    def default(self, o):
+        json_dict = {}
+        public_method_names = [
+            method
+            for method in dir(o)
+            if callable(getattr(o, method))
+            if not method.startswith("_") and method not in ["soup", "links"]
+        ]
```

Comment:
Could you add a test case that checks the list of public methods reported for a scraper object?

Reply:
i think we should add some test for all of the json serialization and methods used within but I would wait until we have decided how to implement this feature.

Reply:
Ok.  I disagree and think that writing a test to confirm the list of expected method names would be useful before the serialization implementation.